src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1493,12 +1493,28 @@ void PolicyHandler::OnSnapshotCreated(const BinaryMessage& pt_string) {
     return;
   }
 
-  AppIdURL app_url = policy_manager_->GetNextUpdateUrl(urls);
-  while (!IsUrlAppIdValid(app_url.first, urls)) {
-    app_url = policy_manager_->GetNextUpdateUrl(urls);
+  const uint32_t app_id = GetAppIdForSending();
+  const std::string app_policy_id =
+      application_manager_.application(app_id)->policy_app_id();
+
+  size_t app_idx = 0;
+  for (; app_idx < urls.size(); ++app_idx) {
+    if ((urls[app_idx].app_id == policy::kDefaultId ||
+         urls[app_idx].app_id == app_policy_id) &&
```

Comment:
@ZhdanovP use style ```const == variable``` for comparing for avoid potential problem with assign (in case of typo you will spend a lot of time to figure out where the problem ```variable = const```)

Reply:
@ZhdanovP I also suggest you to consider helpers::Compare template while comparing two or more values for the same variable